actionpack/lib/action_controller/metal/rescue.rb

```diff
@@ -7,10 +7,8 @@ module Rescue
     include ActiveSupport::Rescuable
 
     def rescue_with_handler(exception)
-      if (exception.respond_to?(:original_exception) &&
-          (orig_exception = exception.original_exception) &&
-          handler_for_rescue(orig_exception))
-        exception = orig_exception
+      if exception.cause && handler_for_rescue(exception.cause)
+        exception = exception.cause
```

Comment:
It'd be interesting to allow rescue handlers for wrapper exceptions as well as their causes. Then a handler could rescue all `TemplateError` and also know to include details about its `#cause`. Separate change to explore.